lib/type/string.ex

```diff
@@ -75,6 +78,43 @@ defmodule Litmus.Type.String do
     {:ok, params}
   end
 
+  @spec regex_validate(t, binary, map) :: {:ok, map} | {:error, binary}
+  defp regex_validate(%__MODULE__{regex: %{pattern: regex, error_message: msg}}, field, params)
+       when regex != nil and msg != nil do
+    if Map.has_key?(params, field) and !Regex.match?(regex, params[field]) do
+      {:error, msg}
+    else
+      {:ok, params}
+    end
+  end
+
+  defp regex_validate(%__MODULE__{regex: %{pattern: regex, error_message: nil}}, field, params)
+       when regex != nil do
+    if Map.has_key?(params, field) and !Regex.match?(regex, params[field]) do
+      {:error, "#{field} must be in a valid format"}
+    else
+      {:ok, params}
+    end
+  end
+
+  defp regex_validate(%__MODULE__{regex: %{pattern: nil}}, _field, params) do
+    {:ok, params}
+  end
+
+  @spec trim(t, binary, map) :: {:ok, map}
+  defp trim(%__MODULE__{trim: true}, field, params) do
+    if Map.has_key?(params, field) do
+      trimmed_params = Map.put(params, field, String.trim(params[field]))
```

Comment:
Maybe let's split `String.trim(params[field])` out into `trimmed_value`, then we can return `{:ok, Map.put(params, field, trimmed_value)}`? That way we don't have nested function calls in the same line.